sway-core/src/parse_tree/declaration/impl_trait.rs

```diff
@@ -51,7 +51,7 @@ impl ImplTrait {
         let impl_keyword = iter.next().unwrap();
         assert_eq!(impl_keyword.as_str(), "impl");
         let trait_name = iter.next().unwrap();
-        assert_eq!(trait_name.as_rule(), Rule::trait_name);
+        assert_eq!(trait_name.as_rule(), Rule::call_path);
```

Comment:
I know these asserts were in here before but I think in general we should be using `debug_assert`. What do you think? I'm open to debate

Reply:
When I was at MS we had this discussion and it was decided that `debug_assert` was a bit toxic.  If we have an invariant then we either check for it in code, or we always assert, since accidentally letting a bug out into the wild without knowing is worse than having it panic and allowing us to fix it.  If the panic in a release build is unpalatable then it should just be a proper check and return a nice error.

This is in the general case.  I'm not really a fan of asserts outside of test code at all, personally.